### ISSUE TITLE
Mutual Auth: Introduce helm feature value `feature.mtls`

### ISSUE THREAD

**Issue by mhofstetter** [cilium/cilium#26772]
Currently, a user has to set multiple helm values to configure mTLS.

e.g. 

* `authentication.mutual.spire.enabled=true`
* `encryption.enabled=true`
* `encryption.type=wireguard`
* ...

Therefore, a Helm value `feature.mtls` (boolean) should be introduced. If set to `true`, the Helm chart should automatically configure the required helm values with reasonable defaults to enable mTLS.

**Comment by meyskens** [comment]
In trying to implement this I hit a lot of edge cases that would not be handled correctly in a simple implementation. 

It would be nice to have a Helm flag for `featureSet` that can be used by other feature sets as well to set multiple flags at once like `featureSet.mtls` or `featureSet.l7`. However these flags need to be overwritten by other flags which we nee to detect in Helm. 
We also need to be able to detect if a previous install had these flags set so we don't overwrite them.
For example we set `encryption.type=wireguard` so we need to prevent the flags `--set feature.mTLS=true --set encryption.type=ipsec --set encryption.enabled=true` will become wireguard not ipsec! 
Also handle upgrades: `helm upgrade --set feature.mTLS=enable --reuse-values` will convert your ipsec to wireguard without you realising it...

The solution would be to use the same tricks we did in https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/clustermesh-apiserver/tls-helm/_helpers.tpl#L12C1-L13 however maintainability becomes a question.
This would mean we have to add a lot of validation and pseudo values for all config flags we want to introduce this way. 

In short I think we're overshooting in this issue to add this for 1.14 and we should come up with a proper agreed upon by all involved maintainers on how to add this very handy user feature without compromising maintainability as well as any configuration regressions caused by people using it.

**Comment by youngnick** [comment]
We are still considering if we will do this or not; pinning to follow up later.

**Comment by youngnick** [comment]
With the deprecation of Mutual Auth (under #47132), this issue is obsolete. Closing to make outstanding issues easier to track.
